chpl/chpl-service/src/main/java/gov/healthit/chpl/web/controller/SurveillanceController.java

```diff
@@ -230,17 +253,40 @@ public void streamDocumentContents(@PathVariable("documentId") Long documentId,
         return new ResponseEntity<Surveillance>(result, responseHeaders, HttpStatus.OK);
     }
 
-    @ApiOperation(value = "Add documentation to an existing nonconformity.",
+    @Deprecated
+    @ApiOperation(value = "DEPRECATED.  Add documentation to an existing nonconformity.",
             notes = "Upload a file of any kind (current size limit 5MB) as supporting "
                     + " documentation to an existing nonconformity. The logged in user uploading the file "
                     + " must have either ROLE_ADMIN or ROLE_ACB and administrative "
                     + " authority on the associated ACB.")
     @RequestMapping(value = "/{surveillanceId}/nonconformity/{nonconformityId}/document/create",
             method = RequestMethod.POST, produces = "application/json; charset=utf-8")
+    public @ResponseBody String uploadNonconformityDocumentDeprecated(@PathVariable("surveillanceId") Long surveillanceId,
+            @PathVariable("nonconformityId") Long nonconformityId, @RequestParam("file") MultipartFile file)
+            throws InvalidArgumentsException, MaxUploadSizeExceededException, EntityRetrievalException,
+            EntityCreationException, IOException {
+
+        return createNonconformityDocumentForSurveillance(surveillanceId, nonconformityId, file);
+    }
+
+    @ApiOperation(value = "Add documentation to an existing nonconformity.",
+            notes = "Upload a file of any kind (current size limit 5MB) as supporting "
+                    + " documentation to an existing nonconformity. The logged in user uploading the file "
+                    + " must have either ROLE_ADMIN or ROLE_ACB and administrative "
+                    + " authority on the associated ACB.")
+    @RequestMapping(value = "/{surveillanceId}/nonconformity/{nonconformityId}/document",
+            method = RequestMethod.POST, produces = "application/json; charset=utf-8")
```

Comment:
This might be a PUT. As in, the intent of the API call is to update a specific Surveillance "thing". That said, we might have to keep it as a POST anyway, as I think file uploads with the Angular tool we have doesn't support PUT. Maybe we have both?

Reply:
I felt this was a post because it was not an idempotent transaction.  Each time you upload the same file, you get another file associated to the nonconformity.